### ISSUE TITLE
Deleting series does not update `numSeries` stat

### ISSUE THREAD

**Issue by jwilder** [influxdata/influxdb#8198]
Using `inmem` index, dropping series does not decrement the `numSeries` stat.  When you restart, it drops and reset correctly.

```
$ influx_stress
$ select numSeries from "monitor"."database" where "database"='stress'
name: database
time                numSeries
----                ---------
1490308160000000000 99931
1490308170000000000 99932
$ drop measurement cpu
(wait 10s or so for stats to be collected)
> select numSeries from "monitor"."database" where "database"='stress'
name: database
time                numSeries
----                ---------
1490308160000000000 99931
1490308170000000000 99932
1490308180000000000 99932
1490308190000000000 99932
```

Stats should have dropped to ~0.

```
$ restart influxd
(wait 10s or so for stats to be collected)
select numSeries from "monitor"."database" where "database"='stress'
name: database
time                numSeries
----                ---------
1490308160000000000 99931
1490308170000000000 99932
1490308180000000000 99932
1490308190000000000 99932
1490308200000000000 99932
1490308210000000000 99932
1490308220000000000 99932
1490308230000000000 99932
1490308240000000000 99932
1490308250000000000 99932
1490308260000000000 99932
1490308270000000000 99932
1490308280000000000 99932
1490308290000000000 99932
1490308300000000000 99932
1490308310000000000 99932
1490308320000000000 99932
1490308340000000000 99932
1490308350000000000 99932
1490308360000000000 99932
1490308380000000000 2
1490308390000000000 2
1490308400000000000 2
```

**Comment by benbjohnson** [comment]
@jwilder Are you still seeing this after 9fb8f1ec1d66c9ac367c4b9dc80738aa72751891? I can't reproduce it.

Also, in your repro steps did you `use stress` before issuing the `DROP MEASUREMENT cpu`?

**Comment by jwilder** [comment]
Yes.  Still occurring w/ d2b396bff58effda5a92eccc734db9d256d404e4

**Comment by jwilder** [comment]
Found the issue.  I'll push up a fix.

**Comment by jwilder** [comment]
Fixed via ee03fbb1
